scripts/generate_refcats_gen3.py

```diff
@@ -0,0 +1,248 @@
+    expression syntax.
+
+    Parameters
+    ----------
+    start, end : `int`
+        The first and last elements of the range, *inclusive*.
+        Assumes ``start <= end``.
+
+    Returns
+    -------
+    range : `str`
+        The Butler syntax for the range.
+    """
+    if start == end:
+        return str(start)
+    else:
+        return f"{start}..{end}"
+
+
+def _to_ranges(ids):
+    """Reexpress a set of integer IDs in terms of Butler dimension
+    expression ranges.
+
+    Parameters
+    ----------
+    ids : iterable [`int`]
+        The IDs to be compressed. Must not contain duplicates.
+
+    Returns
+    -------
+    ranges : iterable [`str`]
+        A collection of Butler range literals and integer literals equivalent
+        to ``ids``. See
+        https://pipelines.lsst.io/modules/lsst.daf.butler/queries.html#literals
+        for syntax.
+
+    Examples
+    --------
+    >>> _to_ranges([1, 2, 3, 5])
+    ["1..3", "5"]
+    """
+    ranges = []
+
+    asc_ids = sorted(ids)
+    range_start = asc_ids[0]
+    # Can replace with itertools.pairwise in Python 3.10
+    for last_id, id in zip(asc_ids, asc_ids[1:]):
+        # Loop invariant: last_id is an element of the contiguous range
+        #     starting with range_start
+        # Loop invariant: range_start <= last_id < id
+        if id > last_id + 1:
+            # last_id ends the range, and id starts a new one
+            ranges.append(_make_range(range_start, last_id))
+            range_start = id
+    ranges.append(_make_range(range_start, id))  # id is last element in asc_ids
+    return ranges
+
+
+id_ranges = _to_ranges(shards)
+
+
+########################################
+# Transfer shards
+
+DEST_DIR = "${AP_VERIFY_HITS2015_DIR}/preloaded/"
+STD_REFCAT = "refcats"
+DEST_RUN = "refcats/imported"
+
+
+def _rename_dataset_type(type, name):
+    """Create a DatasetType that differs from an existing one in name only.
+
+    Parameters
+    ---------
+    type : `lsst.daf.butler.DatasetType`
+        The type to rename.
+    name : `str`
+        The new name to adopt.
+
+    Returns
+    -------
+    new_type : `lsst.daf.butler.DatasetType`
+        The new DatasetType.
+    """
+    return DatasetType(name, type.dimensions, type.storageClass, type.parentStorageClass)
+
+
+src_repo = Butler(args.src_dir, collections=args.src_collection, writeable=False)
+dest_repo = Butler(DEST_DIR, run=DEST_RUN, writeable=True)
+
+
+def _remove_refcat_run(butler, run):
+    """Remove a refcat run and any references from a repository.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.butler.Butler`
+        The repository from which to remove ``run``.
+    run : `str`
+        The run to remove, if it exists.
+    """
+    try:
+        refcat_runs = butler.registry.getCollectionChain(STD_REFCAT)
+        if DEST_RUN in refcat_runs:
+            new_runs = list(refcat_runs)
+            new_runs.remove(DEST_RUN)
```

Comment:
Here and below you're using `DEST_RUN` and not the `run` arg, which doesn't appear to be used here.